### PR TITLE
feat: add Mac OS and Windows support for deeplinks

### PR DESCRIPTION
**Pull request by dshukertjr**
This PR uses [app_links](https://pub.dev/packages/app_links) package instead of [uni_links](https://pub.dev/packages/uni_links) to support all platforms except linux. For linux support, there is an open issue [here](https://github.com/llfbandit/app_links/issues/20). 

This PR should be merged after [this](https://github.com/supabase-community/supabase-flutter/pull/124).

Related https://github.com/supabase-community/supabase-flutter/issues/106

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -69,7 +70,25 @@ void main() async {
 
 ## Authentication
 
-Using authentication can be done easily.
+Using authentication can be done easily.　Using this package automatically persists the auth state on local storage. 
+It also helps you handle authentication with deeplink from 3rd party service like Google, Github, Twitter...
+
```
Comment (bdlukaa) [inline README.md:75]: duplicated line

File: lib/src/supabase_auth.dart [hunk 2]
```diff
@@ -21,12 +27,25 @@ class SupabaseAuth {
   /// {@macro supabase.localstorage.accessToken}
   Future<String?> get accessToken => _localStorage.accessToken();
 
+  /// Returns when the initial session recovery is done.
+  /// Can be used to determine whether a user is signed in or not upon
+  /// initial app launch.
```
Comment (bdlukaa) [inline lib/src/supabase_auth.dart:32]:
```suggestion
  /// Returns when the initial session recovery is done.
  ///
  /// Can be used to determine whether a user is signed in upon initial 
  /// app launch.
```

File: README.md [hunk 3]
```diff
@@ -171,66 +176,186 @@ Follow the guide https://supabase.io/docs/guides/auth#third-party-logins
...
+  For more info: https://developer.apple.com/documentation/xcode/defining-a-custom-url-scheme-for-your-app
+
+</details>
+
+## Windows
+
+<details>
+  <summary>How to setup</summary>
+
+  Setting up deeplinks in Windows has few more steps than other plantforms.
```
Comment (bdlukaa) [inline README.md:254]:
```suggestion
  Setting up deeplinks in Windows has few more steps than other platforms. [Learn more](https://pub.dev/packages/app_links#windows)
```

File: README.md [hunk 4]
```diff
@@ -171,66 +176,186 @@ Follow the guide https://supabase.io/docs/guides/auth#third-party-logins
...
 
-The `android:host` attribute is optional for Deep Links.
+  ...
+  ```
 
-For more info: https://developer.android.com/training/app-links/deep-linking
+  At this point, you can register your own scheme.  
+  On Windows, URL protocols are setup in the Windows registry.
 
-### For iOS
```
Comment (bdlukaa) [inline README.md:202]: Why was "For iOS" removed?

**Review by bdlukaa — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (dshukertjr) [reply README.md:202]: Sorry, it is just moved slightly. 
https://github.com/supabase-community/supabase-flutter/pull/136/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R211

**Comment by dshukertjr** [comment]
@bdlukaa 
Thanks finding all the details that I missed! I have edited and pushed my updates!

**Comment by CodeCakePie** [comment]
Looking forward to this!
Great work @dshukertjr

**Review by bdlukaa — APPROVED** [review]
lgtm

**Pull request merged**